src/ListContainer.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+const flexContainerGallery = {
+  display: 'flex',
+  width: '100%',
+  flexDirection: 'row',
+  flexWrap: 'wrap',
+  justifyContent: 'flex-start',
+  listStyle: 'none'
+};
+
+const flexContainerList = {
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  flexFlow: 'wrap',
+  justifyContent: 'center'
+};
+
+const gridContainerTile = {
+  width: '100%',
+  display: 'grid',
+  gridColumnGap: '0px',
+  gridTemplateColumns: 'auto auto auto auto',
+  justifyContent: 'center'
+};
+
+const ListContainer = ({ display, children}) => {
+  switch (display) {
+    case 'tile':
+      return <ul style={gridContainerTile}>{children}</ul>;
+
+    case 'gallery':
+      return <ul style={flexContainerGallery}>{children}</ul>;
+
+    case 'listing':
+      return <ul style={flexContainerList}>{children}</ul>;
+
+    default:
+      return <ul style={flexContainerList}>{children}</ul>;
+  }
+}
+
+export default ListContainer;
```

Comment:
I moved this into it's own component because what we want to achieve is "single purpose" components. I gave the main `RedditPosts` (now `index`) component the purpose of fetching the posts and creating the state. Which means that now we can pass the reddit posts into any component we want, it's now agnostic of what the data actually looks like.

The purpose of this `ListContainer` component is to determine the styles and wrap it's children in the appropriate container, and presto you get the desired affect, **no matter the children**. This sets us up for providing more flexibility (which I will be adding tomorrow) in how you would let consumers display the content, ie. Grid view, but using `RedditListing` as the display component.

Reply:
Nice I was trying to find out how to do that. But I was told inheritance is not usually wanted in React. 

I liked the whole `{children}` and `{...props}` attributes. I did not know about them before. 
Any more reading you can point me to about that?

Reply:
Children are documented a bit in the React docs, unfortunately you have to kind of hit a point where you think "I just want to wrap the components with something and have it spit out the same components" to go searching for something like using `{children}`, but here's an article my Max https://mxstbr.blog/2017/02/react-children-deepdive/

The `{...props}` is something to use carefully, as it's quite ambiguous in exactly what props you're passing, but useful if you know for sure that every prop needs to be passed on. It's simply using the Spread operator on React components: https://davidwalsh.name/spread-operator

Reply:
Just read that last one. Composition vs Inheritance is different cause in composition your using other objects "baked-in" to your "main" object is how I think of it. 
You can use the other pieces for other things or standalone. 
Inheritance it's getting it's underlying structure from it's parent. 
Think about making a cake with ingredients and then the cake can have different forms and functions depending on the event. 

All the ingredients are objects. 
And a wedding cake can inherit from a standard cake, etc.